docs/DeveloperGuide.adoc

```diff
@@ -295,60 +295,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* * *` |Visually inclined user |Attach an image to each record |Conveniently record additional details of the expense instead of typing it all out
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* * *` |user |Delete a record |Remove expenses/income that I no longer need
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
+|`* * *` |Careless user |Edit an existing record |Make changes to any mistakes made
+
+|`* * *` |user |View a record | View additional information of an expense/income
+
+|`* * *` |user |Tag a record | Categorize and better manage my records
+
+|`* * *` |user |Specify a time stamp on expenses|Know when I spend my money
+
+|`* * *` |user | Sort and filter records by category or tag | Know how my spending/income is distributed
+
+|`* * *` |user | Archive past records | Better manage current expenses
+
+|`* * *` | user | Unarchive records |
+
+|`* * *` |user | View the list of records in an archive | Keep track of what records I have in a particular archive
+
+|`* * *` |user | View the list of all archives | Keep track of how many archives I have and what their names are
+
+|`* * *` |user | Delete an archived record | Delete archive entries that are no longer needed
+
+|`* * *` |user | Deleted an entire archive | Delete any unneeded archive
+
+|`* *` |user | Create custom tags | To better categorise my expenses
+
+|`* *` |user | Hide records | Maintain privacy on certain sensitive expenses/income
+
+|`* *` |Student who tends to overspend | Set a time-based budget | Regulate my spending for the day/week/month with a set limit
+
+|`* *` |Busy student | Automate archiving of outdated records | Not need to manually archive them constantly
+
+|`* *` |Frequent traveller | Add records in foreign currency | Track my expenses when overseas conveniently without having to manually convert currencies
+
+|`* *` |Data oriented user | Have monthly statistics on my expenses | Better track and manage my expenses
+
+|`* *` |User who manages my finance daily | Use one-shot/shortcut commands | Use the program more efficiently
+
+|`*` | User | Set a password for this application | Hide my data
 |=======================================================================
 
 _{More to be added}_
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `Financial Tracker` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== UC01: Adding a tag to an expense
+
+*MSS*
+
+1.  User requests to list expenses
```

Comment:
Do you think that the listing step is redundant? What if the user already knows the index of the expense he wants to tag?